castle/cms/shield.py

```diff
@@ -67,6 +70,9 @@ def protect(req, recheck=False):
                     anonymous = True
         else:
             anonymous = api.user.is_anonymous()
-
         if anonymous:
-            raise Redirect(login_url)
+            access_setting = registry.get('plone.request_access', False)
+            if access_setting:
+                pass
```

Comment:
IF should protect, IF user is anonymous, IF the access_setting is TRUE, then let the user see whatever page they are requesting?

Seems like that is the logical flow of this code change, unless I'm misreading it -- I don't think we want anon users seeing anything if "protect" is enabled. If they are anonymous and getting to the registry page, then the ISecureLoginAllowedView check above should be what allows them to see anything, right?

Reply:
but don't we want anonymous users to get through? because won't users be anonymous until they are granted access and the page they are allowed to see is the page where they request access?